zenoh-kotlin/src/commonMain/kotlin/io/zenoh/sample/Sample.kt

```diff
@@ -30,13 +31,15 @@ import org.apache.commons.net.ntp.TimeStamp
  * @property value The [Value] of the sample.
  * @property kind The [SampleKind] of the sample.
  * @property timestamp Optional [TimeStamp].
+ * @property qos The Quality of Service settings used to deliver the sample.
  * @property attachment Optional [Attachment].
  */
 class Sample(
     val keyExpr: KeyExpr,
     val value: Value,
     val kind: SampleKind,
     val timestamp: TimeStamp?,
+    val qos: QoS,
```

Comment:
Here you can add a default value, in order to not needing to call `QoS.default()` wherever you create a new sample.

```suggestion
    val qos: QoS = QoS.default(),
```

Reply:
The issues is, there is no really default value of qos for sample, since it would depend on the context where sample is used (either in subscribe, or query.reply). It is to be addressed later in the sample rework.